bam_qc_metrics/bam_qc.py

```diff
@@ -101,50 +111,70 @@ def __init__(self, config):
         (self.tmpdir, self.tmp_object) = self.setup_tmpdir(config[self.CONFIG_KEY_TEMP_DIR])
         self.unmapped_excluded_reads = None
         # apply quality filter (if any); get input path for downsampling
+        self.logger.info("Started bam_qc processing")
         result = self.apply_mapq_filter(config[self.CONFIG_KEY_BAM])
         (fast_finder_input_path, self.qual_fail_reads, self.unmapped_excluded_reads) = result
         # find 'fast' metrics on full dataset -- after filtering, before downsampling
+        self.logger.info("Started computing fast bam_qc metrics")
         fast_finder = fast_metric_finder(fast_finder_input_path,
                                          self.reference,
                                          self.expected_insert_max,
-                                         self.n_as_mismatch)
+                                         self.n_as_mismatch,
+                                         self.logger)
         self.fast_metrics = self.update_unmapped_count(fast_finder.metrics)
+        self.logger.info("Finished computing fast bam_qc metrics")
         # apply downsampling (if any)
         if self.sample_rate != None and self.sample_rate > 1:
             slow_finder_input_path = self.generate_downsampled_bam(fast_finder_input_path,
```

Comment:
As you are logging when no down-sampling occurs, also log what the down sample rate is when it does occur.

Reply:
Already being logged here: https://github.com/oicr-gsi/bam-qc-metrics/blob/29880cf5b6556042d6e5835d6fbf488f4199f05a/bam_qc_metrics/bam_qc.py#L233